crates/common/validator/src/validator.rs

```diff
@@ -0,0 +1,39 @@
+use anyhow::{Result, ensure};
+use ream_consensus::{
+    constants::SLOTS_PER_EPOCH, electra::beacon_state::BeaconState,
+    misc::compute_start_slot_at_epoch,
+};
+
+pub fn check_if_validator_active(state: &BeaconState, validator_index: u64) -> Result<bool> {
+    let validator = &state.validators[validator_index as usize];
+    Ok(validator.is_active_validator(state.get_current_epoch()))
+}
```

Comment:
```suggestion
pub fn check_if_validator_active(
    state: &BeaconState,
    validator_index: u64,
) -> anyhow::Result<bool> {
    state
        .validators
        .get(validator_index as usize)
        .map(|validator| validator.is_active_validator(state.get_current_epoch()))
        .ok_or_else(|| anyhow!("Validator index out of bounds"))
}
```

we might as well inline this as there is no reason to declare a variable in this case